frontend/src/utilities/__tests__/router.spec.ts

```diff
@@ -0,0 +1,69 @@
+import { testHook } from '~/__tests__/unit/testUtils/hooks';
+import { setQueryArgument, removeQueryArgument } from '~/utilities/router';
+
+// Mock the NavigateFunction
+const navigate = jest.fn();
+
+describe('setQueryArgument', () => {
+  it('should set a query argument and navigate', () => {
+    // Define the key and value for the query parameter
+    const key = 'param';
+    const value = 'example';
+    // Ensure the query parameter does not exist initially
+    const initialUrl = new URL(window.location.href);
+    initialUrl.searchParams.set(key, value);
+    // Call the hook to set the query parameter
+    testHook(setQueryArgument)(navigate, key, value);
+    // Check if navigate is called with the expected URL
+    const expectedUrl = `${initialUrl.pathname}?${key}=${value}`;
+    expect(navigate).toHaveBeenCalledWith(expectedUrl, { replace: true });
+  });
+  it('should not navigate if query argument is already set to the same value', () => {
+    // Define the key and value for the query parameter
+    const key = 'param';
+    const value = 'example';
+    // Ensure the query parameter is already set to the same value initially
+    const initialUrl = new URL(window.location.href);
+    initialUrl.searchParams.set(key, value);
+
+    // Simulate an existing query parameter with the same value
+    window.history.pushState({}, '', `?${key}=${value}`);
+    // Call the hook to set the query parameter with the same value
+    testHook(setQueryArgument)(navigate, key, value);
```

Comment:
`setQueryArgument` is a simple function. Do not use `testHook`.

Reply:
Thanks, @christianvogt, for pointing this out. I apologize for this oversight.